public/static/plugins/perf/js/trend_chart.js

```diff
@@ -131,7 +131,7 @@ mciModule.factory('PerfChartService', function() {
   // thread `level` and series `item`
   function getValueForAllLevels(level) {
     return function(item) {
-      return item.threadResults[level.idx][cfg.valueAttr]
+      return item.threadResults.hasOwnProperty(level.idx) ? item.threadResults[level.idx][cfg.valueAttr] : null
```

Comment:
In my opinion data normalization is better approach. This code is ok too, however I would ask you to reduce line length. I prefer
```
cond
  ? A
  : B
```
Also, do you have any reason to use `hasOwnProperty`? Usually, `if (item.threadResults[level.idx]) { ` should be enough.

Reply:
The thread levels could change in the future, so normalization might be an issue.

I think I used  hasOwnProperty as a cut and paste from the previous case (where 0 would have been false) so the shorter form should be enough.